src/daemon/rpc_command_executor.cpp

```diff
@@ -186,10 +186,25 @@ namespace {
     }
     return s + " " + (ago < 0s ? "in the future" : "ago");
   }
+
   std::string get_human_time_ago(std::time_t t, std::time_t now, bool abbreviate = false) {
     return get_human_time_ago(std::chrono::seconds{now - t}, abbreviate);
   }
 
+  std::string get_human_time_ago(std::optional<uint64_t> t, std::time_t now, bool abbreviate = false) {
+    std::time_t time;
+    if (t.has_value())
+    {
+      time = t.value() / 1000000000;
```

Comment:
was looking for a constant for this but couldn't find one, there's probably a better way huh?

Reply:
That value isn't nanoseconds, it's just a unix time (i.e. integer seconds since 1970).

The `optional` part of it just makes it not included if you aren't an admin (e.g. if you call `get_info` on a public RPC node it won't get set at all).  So easiest here would be just to use `.value_or(0)` so that you always get a concrete value out of the optional, and avoid needing the overload, something like this:

```C++
  else if (auto last_lokinet_ping = res.last_lokinet_ping.value_or(0);
           last_lokinet_ping < std::time(nullptr) - 60 && !force_registration)
  {
    tools::fail_msg_writer() << "Unable to prepare registration: this daemon has not received a ping from lokinet "
                             << (last_lokinet_ping == 0 ? "yet" : "since " + get_human_time_ago(last_lokinet_ping, std::time(nullptr)));
    return false;
  }
```
and similarly for the last ss ping.

Reply:
yea if you guys are ok with variable assignments in conditions this is probably be best/cleanest

Reply:
This ends up throwing a warning comparing integers of different signs if using the auto keyword (last_lokinet_ping < (time(nullptr) - 60)).

Perhaps an unsigned long would be better here instead of auto?

Reply:
> yea if you guys are ok with variable assignments in conditions this is probably be best/cleanest

The `if (init; cond) { ... }` is a nifty new C++17 thing, completely fine with it.

> This ends up throwing a warning comparing integers of different signs

This ought to work:
```C++
else if (auto last_lokinet_ping = static_cast<std::time_t>(res.last_lokinet_ping.value_or(0));
         last_lokinget_ping < ...
```